mautic/mautic
# Fix campaign looping issue

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes https://github.com/mautic/mautic/issues/11061



#### Description:

Ugly looping issue while campaign rebuild use with `Allow contacts to restart the campaign `



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2.  Add to segment two contacts by filter not manually
3. Create campaign with segment
4. Rebuild segment 
5. Then manually remove from campaign one contact
6. Then rebuild campaign 
7. See rebuild looping
8. After PR should work properly




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11318"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
